infocmdb/attribute.go

```diff
@@ -336,3 +338,388 @@ func (c *Client) GetCiAttributeValueCi(ciId int, attributeName string) (value st
 func (c *Client) UpdateCiAttribute(ci int, ua []v2.UpdateCiAttribute) (err error) {
 	return c.v2.UpdateCiAttribute(ci, ua)
 }
+
+type GetAttributeGroupIdValue struct {
+	Data []struct {
+		GroupId int `json:"id,string"`
+	} `json:"data"`
+}
+
+func (c *Client) GetAttributeGroupIdByName(attributeGroupName string) (attGroupId int, err error) {
+	if err = c.v2.Login(); err != nil {
+		return
+	}
+
+	params := map[string]string{
+		"argv1": attributeGroupName,
+	}
+
+	response := GetAttributeGroupIdValue{}
+	err = c.v2.Query("int_getAttributeGroupIdByAttributeGroupName", &response, params)
+	if err != nil {
+		err = utilError.FunctionError(err.Error())
+		log.Error("Error: ", err)
+		return
+	}
+
+	switch len(response.Data) {
+	case 0:
+		err = utilError.FunctionError(attributeGroupName + " - " + v2.ErrNoResult.Error())
+	case 1:
+		attGroupId = response.Data[0].GroupId
+	default:
+		err = utilError.FunctionError(attributeGroupName + " - " + v2.ErrTooManyResults.Error())
+	}
+
+	return
+}
+
+type respCreateAttributeGroup struct {
+	Success bool         `json:"success"`
+	Message string       `json:"message"`
+	Data    []responseId `json:"data"`
+}
+
+type AttributeGroupParams struct {
+	Name                   string
+	Description            string
+	Note                   string
+	OrderNumber            int
+	ParentAttributeGroupId int
+	IsDuplicateAllow       int
+	IsActive               int
+	userId                 int
+}
+
+func (c *Client) NewAttributeGroupParams() (params *AttributeGroupParams) {
+	params = &AttributeGroupParams{
+		Name:                   "",
+		Description:            "",
+		Note:                   "",
+		OrderNumber:            0,
+		ParentAttributeGroupId: 0,
+		IsDuplicateAllow:       0,
+		IsActive:               1,
+		userId:                 0,
```

Comment:
I don't think the private `userId` field is used anywhere other than including it in the request.
What user is logged/historiced to create the attribute group (and attribute below)?
Why do we need to send it but are not able to provide a user?
Would it make sense to default to the userId from the workflow params somehow?

Reply:
I made the field user id public, so the the UserId can be set, if its still 0, it will take the user_id from the workflow parameters.
I added a function in workflow.go that returns the user_id value.

func (c *Client) GetWorkflowUserId()(userid int,err error)